FreeRTOS-Plus/Test/CBMC/proofs/Socket/vSocketBind/ALLOW_ETHERNET_DRIVER_FILTERS_PACKETS/vSocketBind_harness.c

```diff
@@ -0,0 +1,69 @@
+/* Standard includes. */
+#include <stdint.h>
+#include <stdio.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "list.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_IP_Private.h"
+#include "FreeRTOS_Sockets.h"
+
+
+void *safeMalloc(size_t xWantedSize) {
+        return nondet_bool() ? malloc(xWantedSize) : NULL;
+}
+
+uint16_t prvGetPrivatePortNumber( BaseType_t xProtocol )
+{
+	uint16_t usResult;
+	return usResult;
+}
+
+BaseType_t xIPIsNetworkTaskReady( void )
+{
+	/* Return true saying that the task is ready. */
+	return pdTRUE;
+}
+
+
+/* Random number generator provided by the application. In our case, CBMC provides
+ * an indeterministic value. */
+BaseType_t xApplicationGetRandomNumber( uint32_t *pulNumber )
+{
+	BaseType_t xResult;
+	return xResult;
+}
+
+/* Memory assignment for FreeRTOS_Socket_t */
+FreeRTOS_Socket_t * ensure_FreeRTOS_Socket_t_is_allocated () {
+	FreeRTOS_Socket_t *pxSocket = malloc(sizeof(FreeRTOS_Socket_t));
+
+	pxSocket->u.xTCP.rxStream = malloc(sizeof(StreamBuffer_t));
+	pxSocket->u.xTCP.txStream = malloc(sizeof(StreamBuffer_t));
+	pxSocket->u.xTCP.pxPeerSocket = malloc(sizeof(FreeRTOS_Socket_t));
+
```

Comment:
Really want to avoid testing the effect of null pointers?  At the very least, if you choose to use malloc in place of malloccanfail/assume nonnull, write a comment saying so an maybe saying why.  Someday someone is going to turn on the flag to make malloc work like malloccanfail, and your proof is going to break, and you want to make is easy for the maintainer to fix.

Reply:
Removed this implementation and started using the one with safe malloc.